app/services/checksum_validator.rb

```diff
@@ -87,30 +102,39 @@ def add_cv_result_for_deletions_in_xml(subentity)
     end
   end
 
-  # exploring a solution for #522
-  # for latest sigcat, compare checksums
-  def validate_signature_catalog
-    results = []
-    signature_catalog_entries.each do |sig_cat_entry|
-      calculated_signature = Moab::FileSignature.new.signature_from_file(sig_cat_entry_path(sig_cat_entry))
-      results << sig_cat_entry.signature.eql?(calculated_signature)
+  def validate_signature_catalog_entry(entry)
+    calculated_signature = Moab::FileSignature.new.signature_from_file(signature_catalog_entry_path(entry))
+    unless entry.signature.eql?(calculated_signature)
+      mismatch_error_data = { file_path: signature_catalog_entry_path(entry).to_s, version: entry.version_id }
+      checksum_results.add_result(AuditResults::MOAB_FILE_CHECKSUM_MISMATCH, mismatch_error_data)
     end
-    results
+  rescue Errno::ENOENT
+    absent_from_moab_data = { manifest_file_path: latest_signature_catalog_path,
+                              file_path: signature_catalog_entry_path(entry).to_s }
+    checksum_results.add_result(AuditResults::FILE_NOT_IN_MOAB, absent_from_moab_data)
   end
 
   def moab_storage_object
     Moab::StorageObject.new(druid, druid_path)
   end
 
-  def signature_catalog_entries
-    moab_storage_object.version_list.last.signature_catalog.entries
-  end
-
   def druid_path
     "#{endpoint.storage_location}/#{DruidTools::Druid.new(druid).tree.join('/')}"
   end
 
-  def sig_cat_entry_path(e)
-    Pathname("#{druid_path}/#{Moab::StorageObject.version_dirname(e.version_id)}/data/#{e.group_id}/#{e.path}")
+  def signature_catalog_entry_path(e)
+    Pathname("#{druid_path}/#{Moab::StorageObject.version_dirname(e.version_id)}/#{DATA}/#{e.group_id}/#{e.path}")
+  end
+
+  def latest_signature_catalog_path
+    "#{moab_storage_object.object_pathname}/#{latest_moab_version.version_name}/#{MANIFESTS}/#{SIGNATURE_XML}"
```

Comment:
similarly:

```ruby
latest_moab_version.version_pathname.join(MANIFESTS, SIGNATURE_XML)
```

Reply:
I like it. Hopefully you don't object to appending a `to_s` to convert the `Pathname` to a `String`?